Source/WebInspectorUI/UserInterface/Views/MultiSidebar.js

```diff
@@ -322,15 +384,65 @@ WI.MultiSidebar = class MultiSidebar extends WI.Sidebar
     {
         this.dispatchEventToListeners(WI.Sidebar.Event.WidthDidChange, {sidebar: event.target, newWidth: event.data.newWidth});
     }
+    
+    _recalculateHeight(newHeight)
+    {
+        console.assert(newHeight);
+
+        // Need to add 1 because of the 1px border-top.
+        newHeight = Math.ceil(Number.constrain(newHeight, this.minimumHeight + 1, this.maximumHeight));
+        this.element.style.height = `${newHeight}px`;
+
+        if (this.collapsed)
+            return;
+        
+        for (const sidebar in this.sidebars) {
+            if (sidebar.selectedSidebarPanel)
+                sidebar.selectedSidebarPanel.needsLayout(WI.View.LayoutReason.Resize);
+        }
+
+        this.dispatchEventToListeners(WI.MultiSidebar.Event.HeightDidChange, {newHeight});
+    }
+    
+    _determineBottomness()
+    {
+        const bottomWidth = 550;
+        const shouldBeOnBottom = WI.tabBar._tabContainer.realOffsetWidth <= bottomWidth;
+        
+        if (shouldBeOnBottom === this._actuallyBottomBar)
+            return;
+
+        this._actuallyBottomBar = !!shouldBeOnBottom;
+        document.body.classList.toggle("has-bottom-sidebar", this._actuallyBottomBar);
```

Comment:
FWIW, i think a container query would be preferable to a window query as the latter has an implicit relationship of "the `#content` will always be at least as wide as the window", which may not necessarily be true in the future, whereas the former only ever deals with `#content` (and even then IMO that's a bit of a layering violation since `WI.Sidebar` shouldn't necessarily know/care about whether it's parent is `#content` or something else)

with that having been said, i actually think this should be handled at the top-level (i.e. `Source/WebInspectorUI/UserInterface/Base/Main.js`) as @rcaliman-apple described like "what kind of situation are we currently in right now" (e.g. `WI.LayoutConstraint.None` vs `WI.LayoutConstraint.Narrow`) which would be piped down into the various views (e.g. similar to `WI.updateDockedState`).  this would ensure that the state management of this is kept in as singular of a place as possible, as well as keeping the cause of any state changes in relative proximity (i.e. all the `Main.*` files instead of it being split between differently named files) and allowing us to reuse existing infrastructure (e.g. we already have `WI._windowResized`, `WI.getMaximumSidebarWidth`, etc.)

Reply:
Since I've chosen a MediaQueryList on the JS side, I've chosen to media query for consistency. However, it should be easy to change if you still want to go with the container query.

Reply:
`#content` is the main container for all the tabs, so everything other than the tab bar, split console, and console prompt should be inside it (which is basically all of the UI)

we definitely could do all this on `#main`/`body` if we really wanted to, but the tab bar, split console, and console prompt are all already full width anyways, so it's probably not necessary :)

Reply:
Moved to `#content`.